packages/core/src/prompts/Mixin.ts

```diff
@@ -0,0 +1,91 @@
+type PromptsDict = Record<string, any>;
+
+export class PromptMixin {
+  /**
+   * Validates the prompt keys and module keys
+   * @param promptsDict
+   * @param moduleDict
+   */
+  validatePrompts(
+    promptsDict: PromptsDict,
```

Comment:
A little inconsistent here with the two types with one defined earlier and the other defined inline.